### ISSUE TITLE
Error is always undefined

### ISSUE THREAD

**Issue by sregg** [drwpow/openapi-typescript#1386]
**Description**

When using `GET` or `POST` from `openapi-fetch`'s `createClient`, the `error` object is always typed as `undefined`.
`data` is typed correctly (i.e. the response type or `undefined`).

**Reproduction**

<img width="477" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/1479283/00615168-7443-49aa-9917-f4cbfe6357cc">

<img width="520" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/1479283/4f8af64c-fe9d-4c08-b036-498fa228f019">

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-fetch/blob/main/CONTRIBUTING.md))

**Comment by sregg** [comment]
If I manually add a 404 response for example in the generated types:
<img width="673" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/1479283/2587a7d0-2c0b-487d-8ffe-7c5258a91814">

Then I get a proper type for `error`:

<img width="532" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/1479283/273cdd35-64ec-4a45-9101-53e570e30c39">

Does that mean that the BE has to specify all the possible error status and the returned types?

**Comment by drwpow** [comment]
That’s correct—every endpoint _should_ have an error or `default` response defined per convention. You can even enforce this with linting tools like Redocly’s CLI. I’ve never consumed a production OpenAPI spec that didn’t specify error shapes.

That said, we could convert it to `unknown`—that would be better than `undefined`—but there’s still the bigger problem of your schema not specifying what errors look like at all.

**Comment by sregg** [comment]
Thanks Drew. I'll check with the BE (Micronaut) team if they can specify a default error.

**Comment by jlost** [comment]
Unknown seems like the correct behavior since you "don't know" what the error looks like. Isn't that still an underlying problem here? Should openapi-typescript assume the schema is 100% perfect and 100% complete, or should it account for gaps like TypeScript itself does, as TypeScript itself does?

**Comment by drwpow** [comment]
I’d be open to a PR that changes the type from `undefined` to `unknown` in TS. That wouldn’t be too disruptive of a change (and wouldn’t change runtime code; only in-editor types)
